### ISSUE TITLE
BUG: WSGI incompatibility, destroying downstream A.I. web apps based on Torch, TensorFlow, Jax, ...

### ISSUE THREAD

**Issue by legel** [numpy/numpy#26039]
### Describe the issue:

WSGI and NumPy are not friends.  Someone who understands both libraries should consider helping out what is probably hundreds to thousands of A.I. developers who'd love to deploy a standard Apache server with standard A.I. tools...

### Reproduce the code example:

```python
sudo systemctl start apache2
# ...with a Flask/Python web app, that imports numpy / torch / tensorflow / jax / ...
```

### Error message:

```shell
[Fri Mar 15 17:54:35.669711 2024] [wsgi:error] [pid 896464:tid 140159483893312] /var/www/ecodash/venv/lib/python3.10/site-packages/pandas/__init__.py:14: UserWarning: NumPy was imported from a Python sub-interpreter but NumPy does not properly support sub-interpreters. This will likely work for most users but might cause hard to track down issues or subtle bugs. A common user of the rare sub-interpreter feature is wsgi which also allows single-interpreter mode.
```
```


### Python and NumPy Versions:

Python 3.10.12
NumPy 1.26.4

### Runtime Environment:

Venv on a Linux Machine

### Context for the issue:

When running an Apache web server, system just completely fucks up loading NumPy -- as well as **all** NumPy-dependent numerical processing libraries including Torch, TensorFlow, and JAX.  

Basically everyone developing an A.I. web app is facing this "rare" issue.

**Comment by ngoldbaum** [comment]
dupe of #24755, the main reason this hasn’t been done is it’s an enormous amount of work (at a minimum, a large scale refactor of numpy’s C code from static python types to heap types).

**Comment by legel** [comment]
Given that there is not sufficient interest and resources to resolve this “the right way”, how might you recommend one proceeds to be able to effectively incorporate NumPy et al. in a web serving environment based on Apache etc.?

I’m not looking for anything fancy or special, just the most sensible basic workaround.  I ask because maybe this is more obvious to you and mentioning notes here could potentially help others.

A few starting thoughts / hypotheses:
- Virtual environments such as conda or docker may be possible to create and isolate from web serving environment and make callable with minimal nonsense 
- In the most extreme scenario, I figured it could be possible to write some kind of a server API running outside of the /var/www/[website]/venv … (possibly in its own virtual env like conda or even on a different machine…) that can basically process NumPY computations for users running Apache WSGI web apps 

“Simplicity is ultimate sophistication”, let me know if any ideas come to mind.

**Comment by rkern** [comment]
The standard solution is to use Gunicorn/Uvicorn just behind your Apache/nginx frontend to manage the numpy-using worker processes.

**Comment by legel** [comment]
Finally implemented this change, happy to say "it works!". 
As well, I think with Gunicorn it's faster too.
Thanks again.

**Comment by alissonrodrigo** [comment]
> Finally implemented this change, happy to say "it works!". As well, I think with Gunicorn it's faster too. Thanks again.

I'm having the same problem. How did you solve it? Thanks
